### DISCUSSION TITLE
Ingredients in two different steps

### DISCUSSION THREAD

**Discussion by MattSidney** [TandoorRecipes/recipes#2224] [General]
Hoping to understand if it is possible to add ingredients to different steps without creating additional ingredients in the main ingredient list. 

For example: If a recipe calls for "lemon zest from 1 lemon" I would want to break it down like the below:

- Main ingredient list:
  - 1 Lemon

- Step 1:
  - Ingredients:
    - 1 Lemon
  - Instructions:
    - zest lemon

- Step 2:
  - Ingredients:
    - Lemon zest
  - Instructions:
    - add lemon zest to XYZ

Another example below:

- Main ingredient list:
  - 1 bunch Parsley

- Step 1:
  - Ingredients:
    - 1 bunch Parsley
  - Instructions:
    - Remove enough leaves to fill 1 cup, loosly packed. Reserve stems. 

- Step 2:
  - Ingredients:
    - 1 cup parsley leaves   
  - Instructions:
    - add Parlsey leaves to XYZ

- Step 3
  - Ingredients:
    - reserved parsley stems
  - Instructions:
    - add reserved parsley stems to ZYX

In these cases the preparation step would essentially "create" a new "ingredient" that can then be used at a later step(s). However, we don't need to purchase that new ingredient. 

Not sure if a flag could be added to the existing ingredient options to "exclude from ingredient list/purchase list"? 
<img width="219" alt="image" src="https://user-images.githubusercontent.com/45581528/208782854-af88b5d7-079a-4017-9767-25e09844515a.png">

There is a bigger more complicated option to allow a recipe to generate a "prepared" ingredients but not sure the additional complication is worth it.

@vabene1111 would love your thoughts on this :)

PS. Of course, this wouldn't restrict the current option to just add preparation information in the Note section but would certainly be nice to have if you want to track ingredients, and prepare ingredients on a per step basis.

**Comment by vabene1111** [comment]
i understand what you mean and its an interesting thought. Recipes usually consist of stuff prepared in a first step and later used in another. 

Honestly while I like the feature I do not really see any way in adding this without adding being a very complicated and hard to understand feature. I usually just add the prepared ingredient into the text and make it bold which serves basically the same purpose. 

Maybe If someone can come up with a good idea on how to communicate and implement this in a clearly understandable way for the user this could be done.
